controllers/component/component_adapter.go

```diff
@@ -52,6 +55,17 @@ func NewAdapter(component *applicationapiv1alpha1.Component, application *applic
 	}
 }
 
+// EnsureComponentHasFinalizer is an operation that will ensure components
+// that have been created are assigned a finalizer
+func (a *Adapter) EnsureComponentHasFinalizer() (controller.OperationResult, error) {
+	if !controllerutil.ContainsFinalizer(a.component, h.ComponentFinalizer) {
+		controllerutil.AddFinalizer(a.component, h.ComponentFinalizer)
```

Comment:
I don't think that this call updates(patches) the resource on the cluster, it only adds the finalizer to the object without applying it.
We should probably have a function like [AddFinalizerToPipelineRun](https://github.com/redhat-appstudio/integration-service/blob/c199d2673bc5c7dc9936798acce1ffd9bad6864b/helpers/integration.go#L468) for this.

Untested idea: maybe we can even make that function generic so we don't have duplicate versions for all different kinds of objects.

Reply:
You're right. Should appropriately patch now in the new helper function in most recent pr.
The PLR and Component finalizer functions share near identical code, but I'm not sure the best way to go about turning the functions into one generic with both ` pipelineRun *tektonv1.PipelineRun`  and  `component *applicationapiv1alpha1.Component` types?